tests/session/test_env_select.py

```diff
@@ -15,6 +15,57 @@
 
 CURRENT_PY_ENV = f"py{sys.version_info[0]}{sys.version_info[1]}"  # e.g. py310
 
+####################################################################
+#   Tests directly on env_select functionality.
+
+
+@pytest.mark.parametrize(
+    ("user_input", "env_names", "is_all", "is_default"),
+    [
+        (None, (), False, True),
+        ("", (), False, True),
+        ("a1", ("a1",), False, False),
+        ("a1,b2,c3", ("a1", "b2", "c3"), False, False),
+        #   If the user gives "ALL" as any envname, this becomes an "is_all" and other envnames are ignored.
+        ("ALL", (), True, False),
+        ("a1,ALL,b2", (), True, False),
+        #   Zero-length envnames are ignored as being not present. This is not intentional.
+        (",,a1,,,b2,,", ("a1", "b2"), False, False),
+        (",,", (), False, True),
+        #   Environment names with "invalid" characters are accepted here; the client is expected to deal with this.
+        ("\x01.-@\x02,xxx", ("\x01.-@\x02", "xxx"), False, False),
+    ],
+)
+def test_clienv(user_input: str, env_names: tuple[str], is_all: bool, is_default: bool) -> None:
+    c = CliEnv(user_input)
```

Comment:
please don't use one character variable names 👍

Reply:
Given that it's the _only_ variable in that entire function, which is only three lines long, making it a longer variable name adds only verbosity. (There's a reason that, even though we now can have very long variable names, we say `for i in ...` rather than `for index_of_list ...`.) But I have changed it as you requested.